src/components/tl/ucp/barrier/barrier_knomial.c

```diff
@@ -0,0 +1,21 @@
+#include "config.h"
+#include "tl_ucp.h"
+#include "barrier.h"
+#include "core/ucc_progress_queue.h"
+
+ucc_status_t ucc_tl_ucp_barrier_knomial_progress(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+    /* TODO implement main logic */
+    task->super.super.status = UCC_OK;
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_ucp_barrier_knomial_start(ucc_coll_req_t *req)
```

Comment:
it looks like that every post function will enqueue task to the queue. Isn't it better to move enqueue to common place, e.g. ucc_collecitve_post?
why argument is of type ucc_coll_req_t?

Reply:
1. argument: does not really matter. i can change it to ucc_coll_task. the key here is that the argument to post would always inherit from ucc_coll_task which in turn from ucc_coll_req_t.
2. I thought that some TL/CL might want to bypass core progress, if, e.g., the task does not need "progress" function. Lets say it runs on some device, ie offloaded AND the task does not have any dependencies. Then you don't need to enqueue and can avoid overhead.

Reply:
1. I think task is better for consistency with progress
2. Don't we need to poll for completion to know status even if there is no actual progress function?

Reply:
the polling for completion is just checking the "status" field. If the TL implements everything (and does not need tasks/schedules) then it can just set COMPLETE flag. No need for queue/dequeue.